app/records/views.py

```diff
@@ -85,6 +87,72 @@ def get_subjects_enrichment(subjects_list: list[str], limit: int = 10) -> dict:
         return {}
 
 
+def get_delivery_options_context(iaid: str) -> dict:
+    """
+    Fetch and process delivery options for a record.
+
+    Args:
+        iaid: The document iaid
+
+    Returns:
+        Dictionary with delivery options context containing:
+        - delivery_option: The AvailabilityCondition name as string (if valid)
+        - availability_group: The availability group name (if mapped to a group)
+        May return empty dict if unavailable or on error.
+    """
+    try:
+        delivery_result = delivery_options_request_handler(iaid)
+
+        # Validate we got results and it's a list
+        if not delivery_result or not isinstance(delivery_result, list):
+            logger.info(f"No delivery options available for iaid {iaid}")
+            return {}
+
+        # Check list is not empty before accessing first element
+        if len(delivery_result) == 0:
```

Comment:
Consider adding a comment to explain why `0` is returned in this case - it would help understand the intent.
What do mean by `empty` in the logger?